cumulusci/utils/yaml/cumulusci_yml.py

```diff
@@ -0,0 +1,41 @@
+from typing import IO, Text
+from re import compile, MULTILINE
+from logging import getLogger
+from io import StringIO
+
+import yaml
+
+NBSP = "\u00A0"
+
+pattern = compile(r"^\s*[\u00A0]+\s*", MULTILINE)
+
+logger = getLogger(__name__)
+
+
+def _replace_nbsp(origdata):
+    counter = 0
+
+    def _replacer_func(matchobj):
+        nonlocal counter
+        counter += 1
+        string = matchobj.group(0)
+        rc = string.replace(NBSP, " ")
+        return rc
+
+    data = pattern.sub(_replacer_func, origdata)
+
+    if counter:
+        plural = "s were" if counter > 1 else " was"
+        logger.warn(
+            f"Note: {counter} non-breaking space character{plural} detected in cumulusci.yml.\n"
```

Comment:
The code is counting the number of lines that matched; not the number of characters.

Reply:
I changed the message to clarify. If this message frequently gets invoked, in a future version, could try to help the user pinpoint to a line number.

Reply:
https://github.com/SFDO-Tooling/CumulusCI/pull/1596/commits/1b455898665ceedec330dea68e53ece4719b2898#diff-9c0b3c5a896ba84e51be4be79b9a8414R30